aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/options/FederateToIdentityPoolOptions.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.options
+
+/**
+ * Options to provide additional fields to federate to Cognito Identity Provider.
+ */
+data class FederateToIdentityPoolOptions internal constructor(val developerProvidedIdentityId: String?) {
```

Comment:
why is `developerProvidedIdentityId` nullable as it the only property?

Reply:
It has to be nullable if we provide a default options:
"I would prefer to pass default options instead of null like all other APIs.". Nullable allows a customer to provide a builder without passing anything.

Reply:
That's true, but that will be inconsistent with other APIs where we provide `Default`. I think this is due to Java not supporting default values (and hence overloaded methods with and without options)
> Nullable allows a customer to provide a builder without passing anything.

Customer should not be using API with options if they don't want to pass anything. But instead use the overloaded one without it.

Reply:
@tylerjroach, since `developerProvidedIdentityId` is part of public API, let's make sure that we consider inconsistencies and Java support.